rls-span/src/lib.rs

```diff
@@ -78,6 +82,62 @@ impl Column<ZeroIndexed> {
     }
 }
 
+impl Step for Column<ZeroIndexed> {
+    fn steps_between(start: &Self, end: &Self) -> Option<usize> {
+        Some((end.0 - start.0) as usize)
+    }
+
+    fn replace_one(&mut self) -> Self {
+        self.0 = 1;
+        self.clone()
+    }
+
+    fn replace_zero(&mut self) -> Self {
+        self.0 = 0;
+        self.clone()
+    }
+
+    fn add_one(&self) -> Self {
+        Self::new(self.0 + 1)
+    }
+
+    fn sub_one(&self) -> Self {
+        Self::new(self.0 - 1)
+    }
+
+    fn add_usize(&self, n: usize) -> Option<Self> {
+        (self.0 as usize).checked_add(n).and_then(|n| u32::try_from(n).ok().map(|n| Self::new(n)))
+    }
+}
+
+impl Step for Column<OneIndexed> {
+    fn steps_between(start: &Self, end: &Self) -> Option<usize> {
+        Some((end.0 - start.0) as usize)
+    }
+
+    fn replace_one(&mut self) -> Self {
+        self.0 = 2;
```

Comment:
yes

Reply:
It's not at all clear what `one` means here though - if it is meant to be the multiplicative identity or an increment, then this is clearly wrong.

Reply:
It looks like it's needed by `RangeInclusive` to construct a "past-the-iterator" value by essentially setting start to `one` and end to `zero` (so only `one > zero` needs to hold apparently)

https://github.com/rust-lang/rust/blob/0c6091fbd0eee290c651f73be899f221eeab3c05/src/libcore/iter/range.rs#L342

So this looks good since we're incrementing both `zero` and `one` by 1